client/modules/_hooks/src/reconportal/useGetOpenTopo.ts

```diff
@@ -0,0 +1,176 @@
+import { useQuery } from '@tanstack/react-query';
+import apiClient from '../apiClient';
+import * as turf from '@turf/turf';
+import { Feature, Point } from 'geojson';
+
+export type OpenTopoFeature = {
+  type: 'Feature';
+  properties: {
+    id: string;
+    name: string;
+    productAvailable: string;
+    dateCreated: string;
+    temporalCoverage: string;
+    doiUrl: string;
+    url: string;
+    host: string;
+    keywords: string;
+    alternateName: string;
+  };
+  geometry: {
+    type: 'Point' | 'Polygon' | 'MultiPolygon';
+    coordinates: number[] | number[][][] | number[][][][];
+  };
+};
+
+export type OpenTopoResponse = {
+  Datasets: {
+    Dataset: {
+      name: string;
+      identifier: {
+        value: string;
+      };
+      alternateName: string;
+      url: string;
+      fileFormat: string;
+      description: string;
+      dateCreated: string;
+      temporalCoverage: string;
+      keywords: string;
+      spatialCoverage: {
+        geo: {
+          geojson: {
+            features: {
+              geometry: {
+                type: 'Polygon' | 'MultiPolygon';
+                coordinates: number[][][] | number[][][][];
+              };
+            }[];
+          };
+        };
+      };
+    };
+  }[];
+};
+
+export type ProcessedOpenTopoData = {
+  original: {
+    type: 'FeatureCollection';
+    features: OpenTopoFeature[];
+  };
+  centroids: {
+    type: 'FeatureCollection';
+    features: OpenTopoFeature[];
+  };
+};
+
+async function fetchOpenTopoData(): Promise<OpenTopoResponse> {
+  const res = await apiClient.get<OpenTopoResponse>('/recon-portal/opentopo/');
+  return res.data;
+}
+
+function preprocessData(data: OpenTopoResponse): ProcessedOpenTopoData {
+  const seenUrls = new Map<string, OpenTopoFeature>();
```

Comment:
We should rework this or see if it makes more sense if we get the center point in the backend or if that is needed at all in the new design.